app/src/main/java/io/homeassistant/companion/android/onboarding/manual/ManualSetupPresenterImpl.kt

```diff
@@ -24,6 +24,10 @@ class ManualSetupPresenterImpl @Inject constructor(
             url = URL(urlString)
         } catch (e: MalformedURLException) {
             Log.e(TAG, "Unable to parse url", e)
+            if(e.message != null){
+                view.onErrorUrlParse(e.message)
```

Comment:
We should use a more friendly message. You can let the view get string from a resource

Reply:
What is the best way to test what the error is in order to display the correct message?

Reply:
Okay where is the best place to have that logic
in the manualSetupPresenter or ManualSetupFragment

Reply:
You will not be able to getString from the presenter (do not use a context inside the presenter) so in the Fragment do something like that 
`view?.findViewById<TextInputLayout>(R.id.url_text_layout)?.error = getString(R.string.error_url)`

Reply:
Do you know if URL() throws any exceptions other than No Protocol that we need to catch and display a custom error message for?

Reply:
The problem that I see here is that it servery limits the Errors that can be shown to the user. right now we show on the No protocol error but there could be other errors in URL Parsing and it will annoy users if they get an irreverent error. 
I think that this solution allows for needed future expandability and communicating as fully as possible what went wrong.

Reply:
We should not rely on MalformedURLException error message as it's from the JVM.
We are not sure about the implementation on each platform (android version).

Reply:
Yeah. We should just have an error message like: Unable to parse your Home Assistant URL. It should look like `https://example.com`.

Reply:
Okay I changed it to what you suggested.